### PR TITLE
Buyground and build on it test

### PR DESCRIPTION
**Pull request by daanwerf**
I have added simple code for the agent to be able to ask to buy land, build on that land and demolish a building. These are some of the basic things that the agent can accomplish without the extra percepts that are needed. This code merge will satisfy (parts of) the following user stories: 

"As an agent When in a 
simulation I am able to buy 
ground if I think this is 
benificial for me. To construct 
things that are part of my goals."

"As an agent
When in a simulation
I am able to destroy stuff if I think 
this is benificial for me.
To make room for construction or to 
optimize a certain indicator."

"As an agent
When in a simulation
I am able to build stuff if I think this 
is benificial for me.
To optimize a certain indicator or to 
reach a construction goal."

### CODE REVIEW

File: TU-Agent/tygron.act2g [hunk 1]
```diff
@@ -2,4 +2,8 @@ use dummy as knowledge.
 
 define	building_plan_construction(Id, Level, MultiPolygon) with
    pre{ true }
-   post{ true }
\ No newline at end of file
+   post{ true }
+
```
Comment (MichielvdBerg) [inline TU-Agent/tygron.act2g]: Maybe some comments to make clear what is defined?

**Comment by jzoon** [comment]
It works well, just two things to improve the code.

First shouldn't the insert(land(1)) in tygron.mod2g be insert(land(n+1)) to get the correct amount of areas you have bought?

Also, in the same file you have a line where you delete the believe 'havebuilding' and then immediately adopt it again. Is this necessary? This is in line 10.

File: TU-Agent/tygron.mod2g [hunk 2]
```diff
@@ -5,8 +5,17 @@ use somegoal as goals.
...
-		then building_plan_construction(Id, 1, square(10,10,200,10)).
-
-}
+	% take down the buildings that we have just built
+	if bel(buildings([building(Id, Name, List, Status)])) 
+        then building_plan_demolish(Id) + delete(buildings([building(Id, Name, List, Status)])) + delete(havebuilding) + adopt(havebuilding).
+	
+	% find a function to build a student building
+	 if bel(functions(FS), member([NAME,Id,L],FS), member('EDUCATION',L), land(1)) 
+		then building_plan_construction(Id, 3, square(10,10,200,10)) + insert(built(1)).
```
Comment (MichielvdBerg) [inline TU-Agent/tygron.mod2g:14]: I think it is better to make clear what you have built. So for example built(education_normalroom) instead of built(1).

**Comment by daanwerf** [comment]
@jzoon I agree on the first part, but I think Michiel's suggestion fits the situation a bit better, so I will incorporate that. On the second part, the deletion and adoption are necassary as the goal for having a building will be met when there are any buildings anywhere on the map. Because we start the map with a buiding on it, this goal is instantly met so in order to make sure the agent does not terminate we need to adopt the goal again. The deletion of the believe is needed to accompany the demolish action, after which the agent should no longer believe that there is a building.

File: TU-Agent/tygron.mod2g (hunk 2 above)
Comment (daanwerf) [inline TU-Agent/tygron.mod2g:14]: @MichielvdBerg Good point, I'll put that in.

**Comment by MichielvdBerg** [comment]
I agree on merging this pull request

**Comment by jzoon** [comment]
Ah, that explains it and it's fine for this moment. I'll merge.

**Pull request merged**